Submitted By:            zippo zippo@oppiz.net
Date:                    2008-10-28
Initial Package Version: 3.8.3
Upstream Status:         Unknown
Origin:                  zippo@oppiz.net
Description:Fixes things up for GCC-4.3.2

diff -Naur id3lib-3.8.3/examples/demo_convert.cpp 32/id3lib-3.8.3/examples/demo_convert.cpp
--- id3lib-3.8.3/examples/demo_convert.cpp	2003-03-02 00:23:00.000000000 +0000
+++ 32/id3lib-3.8.3/examples/demo_convert.cpp	2008-10-29 20:10:04.000000000 +0000
@@ -84,7 +84,7 @@
   }
 }
 
-int main( unsigned int argc, char * const argv[])
+int main(int argc, char * const argv[])
 {
   flags_t ulFlag = ID3TT_ALL;
   gengetopt_args_info args;
diff -Naur id3lib-3.8.3/examples/demo_copy.cpp 32/id3lib-3.8.3/examples/demo_copy.cpp
--- id3lib-3.8.3/examples/demo_copy.cpp	2003-03-02 00:23:00.000000000 +0000
+++ 32/id3lib-3.8.3/examples/demo_copy.cpp	2008-10-29 20:10:54.000000000 +0000
@@ -81,7 +81,7 @@
   }
 }
 
-int main( unsigned int argc, char * const argv[])
+int main(int argc, char * const argv[])
 {
   int ulFlag = ID3TT_ID3;
   ID3D_INIT_DOUT();
diff -Naur id3lib-3.8.3/examples/demo_info.cpp 32/id3lib-3.8.3/examples/demo_info.cpp
--- id3lib-3.8.3/examples/demo_info.cpp	2003-03-02 00:23:00.000000000 +0000
+++ 32/id3lib-3.8.3/examples/demo_info.cpp	2008-10-29 20:09:31.000000000 +0000
@@ -309,7 +309,7 @@
 
 #define DEBUG
 
-int main( unsigned int argc, char * const argv[])
+int main(int argc, char * const argv[])
 {
   ID3D_INIT_DOUT();
 
diff -Naur id3lib-3.8.3/examples/demo_tag.cpp 32/id3lib-3.8.3/examples/demo_tag.cpp
--- id3lib-3.8.3/examples/demo_tag.cpp	2003-03-02 00:23:00.000000000 +0000
+++ 32/id3lib-3.8.3/examples/demo_tag.cpp	2008-10-29 20:10:26.000000000 +0000
@@ -46,7 +46,7 @@
     os << "v2";
 }
 
-int main( unsigned int argc, char * const argv[])
+int main(int argc, char * const argv[])
 {
   int ulFlag = ID3TT_ID3;
   ID3D_INIT_DOUT();
diff -Naur id3lib-3.8.3/include/id3/id3lib_strings.h 32/id3lib-3.8.3/include/id3/id3lib_strings.h
--- id3lib-3.8.3/include/id3/id3lib_strings.h	2003-03-02 00:23:00.000000000 +0000
+++ 32/id3lib-3.8.3/include/id3/id3lib_strings.h	2008-10-29 20:03:28.000000000 +0000
@@ -30,6 +30,7 @@
 #define _ID3LIB_STRINGS_H_
 
 #include <string>
+#include "string.h"
 
 #if (defined(__GNUC__) && (__GNUC__ >= 3) || (defined(_MSC_VER) && _MSC_VER > 1000))
 namespace std
diff -Naur id3lib-3.8.3/include/id3/writers.h 32/id3lib-3.8.3/include/id3/writers.h
--- id3lib-3.8.3/include/id3/writers.h	2003-03-02 00:23:00.000000000 +0000
+++ 32/id3lib-3.8.3/include/id3/writers.h	2008-10-29 20:05:56.000000000 +0000
@@ -30,7 +30,7 @@
 
 #include "id3/writer.h"
 #include "id3/id3lib_streams.h"
-//#include <string.h>
+#include "string.h"
 
 class ID3_CPP_EXPORT ID3_OStreamWriter : public ID3_Writer
 {
diff -Naur id3lib-3.8.3/src/field.cpp 32/id3lib-3.8.3/src/field.cpp
--- id3lib-3.8.3/src/field.cpp	2003-03-02 00:23:00.000000000 +0000
+++ 32/id3lib-3.8.3/src/field.cpp	2008-10-29 20:02:19.000000000 +0000
@@ -33,6 +33,8 @@
 #include "readers.h"
 #include <assert.h>
 
+#include "string.h"
+
 using namespace dami;
 
 // This is used for unimplemented frames so that their data is preserved when
